2020 standards updates

dmallory opened the pull request:
This is intended to capture the decisions we made as a team a year ago now to reflect what's important to us for 2020. It's not meant to re-open the debate, though doing that again for 2021 is certainly valid. I just want to finally get what we already agreed on in as a starting point so we can close those out. To that end please suggest specific changes (grammar, spelling, etc.) or provide alternate wording if it's just clarity and accuracy, and I'll either take what's suggested or what has the most votes if there are more than one to keep this moving once I'm back from leave. Bring opinions on further changes or going back on what we decided to some other event that isn't this PR. Thank you.

dmallory committed:
Removals

dmallory committed:
Extensions

dmallory committed:
Organization

dmallory committed:
Spacing

dmallory committed:
Comments

dmallory committed:
Types

dmallory committed:
Switch

gpambrozio commented on README.markdown:
```diff
@@ -370,28 +341,32 @@ func reticulateSplines(spline: [Double]) -> Bool {
-                       adjustmentFactor: Double,
-                       translateConstant: Int,
-                       comment: String) -> Bool {
+func reticulateSplines(
+  spline: [Double],
+  adjustmentFactor: Double,
+  translateConstant: Int,
+  comment: String
+) -> Bool {
```
nitpick: have 4 spaces to reflect our standard

gpambrozio commented on README.markdown:
```diff
@@ -370,28 +341,32 @@ func reticulateSplines(spline: [Double]) -> Bool {
-                  gender: .male,
-                  location: "San Francisco, CA",
-                  occupation: .doctor)
+let john = Person(
+  fullName: "John Doe",
+  gender: .male,
+  location: "San Francisco, CA",
+  occupation: .doctor
+)
```
nitpick: have 4 spaces to reflect our standard

gpambrozio commented on README.markdown:
```diff
@@ -41,7 +41,6 @@ Descriptive and consistent naming makes software easier to read and understand.
 - use camel case (not snake case)
 - use uppercase for types and protocols, lowercase for everything else
 - name functions and parameters based on role, not on type
-- strive to name functions in a way that reads like a sentence
```
Are we not trying to do this? I at least think it's great when a method follow this.

dmallory replied:
We voted to get rid of this, after discussion of the really long and verbose type and function names, differing to the more general guidance on naming instead.

gpambrozio commented on README.markdown:
```diff
@@ -204,15 +177,13 @@ class TestDatabase : Database {
 
 When they are needed, use inline comments to explain **why** a particular piece of code does something. Comments must be kept up-to-date or deleted.
 
-Avoid block comments inline with code, as the code should be as self-documenting as possible.
```
I think this is still a good statement to keep. It doesn't say to not use it, just to try to avoid it.

dmallory replied:
We also voted to get rid of this after some discussion that inline comments have a place and should be used with judgement that didn't warrant a rule against it.

gpambrozio commented on README.markdown:
```diff
@@ -1023,7 +938,7 @@ func updateCurrentUser(notification: Notification) {
       // Since `updateCurrentUser` is not guaranteed to be called on the main thread,
       // We must dispatch onto the main thread before calling `mt_` functions.
-      DispatchQueue.main.async{ 
+      DispatchQueue.main.async{
```
nitpick: add space before `{` to conform with standards

dmallory replied:
I removed these examples related to `mt_` since they no longer apply without the rules.

dmallory committed:
Change example code to 4 space indents

dmallory committed:
Remove mt_ examples

dmallory commented:
Even though it'll diverge us a lot more on the fork I went ahead and updated all the example code to 4 spaces since I don't see us ever making a lot of effort to stay synced with the original guide. I also removed the additional `mt_` examples. For the others, I'm sticking to the decisions we voted on in early 2020 and not intending this PR to revisit them, just implement them per the spreadsheet votes we recorded and the stories created out of that.

dmallory commented:
@jemersondox @lintmachine @jasdeepsaini please give feedback if you have any or approve this, it's been open for a month and I'd like to get it updated. Also adding @NormDoximity so that someone else continuing on dialer is part of it. Thanks.

NormDoximity commented on README.markdown:
```diff
@@ -366,32 +336,36 @@ Keep short function declarations on one line including the opening brace:
-                  location: "San Francisco, CA",
-                  occupation: .doctor)
+    let john = Person(
+    fullName: "John Doe",
+    gender: .male,
+    location: "San Francisco, CA",
+    occupation: .doctor
```
should these parameters be indented?

dmallory replied:
Good catch, I'll update that one.

NormDoximity reviewed (approved):
I just had one minor question, other than that, everything looks correct based on my recollection of past standards discussions.

jasdeepsaini commented:
@dmallory  Sorry lost track of this PR.  I will take a look later today.

dmallory committed:
Fix indent
